src/Dialogs/ChangePasswordDialog.vala

```diff
@@ -103,7 +103,7 @@ public class SwitchboardPlugUserAccounts.ChangePasswordDialog : Gtk.Dialog {
         action_area.show_all ();
 
         pw_editor.validation_changed.connect (() => {
-            if (pw_editor.is_valid && is_authenticated) {
+            if (pw_editor.is_valid && get_permission ().allowed || pw_editor.is_valid && pw_editor.is_obscure && is_authenticated) {
```

Comment:
I would advise checking if `get_permission` doesn't return `null` first and then checking it if's allowed.
Also I would put all of these checks in separate lines at this point.